pandas/core/internals/blocks.py

```diff
@@ -208,6 +216,12 @@ def internal_values(self):
         """
         return self.values
 
+    def array_values(self) -> ExtensionArray:
```

Comment:
do we need external_values at all? this is getting very confusing with all of the *values*

Reply:
> do we need external_values at all? this is getting very confusing with all of the values

I clarified the docstrings (in this PR, but split off and already merged in #31103) to make it clear what `external_values` (-> `.values`), `internal_values` (-> `_values`) and `array_values` (-> `.array`) are used for. As long as those all have different behaviour (something that can be gradually cleaned up, but not here), we need a mechanism to create those different results.